### ISSUE TITLE
No values in a repeatable field makes file unreadable

### ISSUE THREAD

**Issue by jimfum** [elastacloud/parquet-dotnet#272]
Spark (using Databricks 3.3, Spark 2.2.0) doesn't read repeatable string fields from parquet file created by parquet-dotnet 1.5.1.

```
new DataSet(..., new SchemaElement<IEnumerable<String>>("all_variables"))
ds.Add(..., new String[]{"str1", "str2"})
```

loading the file in Spark:

```
df = spark.read.parquet("/FileStore/tables/test.parquet")
display(df)
```

results in:

schema correctly read and displaying:
```
df:pyspark.sql.dataframe.DataFrame
  ...
  all_variables:array
    element:string
```
but the data displayed **show just "[]"** instead of the two strings expected.

Anything I'm missing?

Thanks for the great work guys!

**Comment by jimfum** [comment]
In addition if I pass in string array with several fields, Databricks Spark shows just [] (empty array) but if I pass in an empty string array (why not), Spark fails with an exception.

`
SparkException: Job aborted due to stage failure: Task 0 in stage 27.0 failed 1 times, most recent failure: Lost task 0.0 in stage 27.0 (TID 27, localhost, executor driver): java.io.IOException: totalValueCount == 0
	at com.databricks.sql.io.parquet.VectorizedColumnReader.<init>(VectorizedColumnReader.java:153)
	at com.databricks.sql.io.parquet.VectorizedColumnReader.getReader(VectorizedColumnReader.java:115)
	at com.databricks.sql.io.parquet.VectorizedFieldReader.getReader(VectorizedFieldReader.java:91)
	at com.databricks.sql.io.parquet.VectorizedArrayReader.getReader(VectorizedArrayReader.java:75)
	at com.databricks.sql.io.parquet.VectorizedFieldReader.getReader(VectorizedFieldReader.java:81)
	at com.databricks.sql.io.parquet.VectorizedFieldReader.getReader(VectorizedFieldReader.java:62)
	at com.databricks.sql.io.parquet.DatabricksVectorizedParquetRecordReader.checkEndOfRowGroup(DatabricksVectorizedParquetRecordReader.java:363)
	at com.databricks.sql.io.parquet.DatabricksVectorizedParquetRecordReader.nextBatch(DatabricksVectorizedParquetRecordReader.java:277)
	at org.apache.spark.sql.execution.datasources.parquet.VectorizedParquetRecordReader.nextKeyValue(VectorizedParquetRecordReader.java:137)
	at org.apache.spark.sql.execution.datasources.RecordReaderIterator.hasNext(RecordReaderIterator.scala:39)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1$$anonfun$prepareNextFile$1.apply(FileScanRDD.scala:240)
	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1$$anonfun$prepareNextFile$1.apply(FileScanRDD.scala:228)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
`

**Comment by aloneguid** [comment]
I can read this in 2.0 preview which you can get from custom nuget feed at https://ci.appveyor.com/nuget/parquet-dotnet-nvq9lkymhrek (unfortunately no time to check or fix in 1.5 as i'm the only one on the project at the moment)

```csharp
      [Fact]
      public void Repeatable_field_writes_reads()
      {
         var ds = new DataSet(new Field<int>("id"), new Field<IEnumerable<string>>("repeats"));
         ds.Add(1, new[] { "one", "two", "three" });

         ParquetWriter.WriteFile(ds, "c:\\tmp\\1.parquet");
      }
```
![image](https://user-images.githubusercontent.com/3155189/33153756-a5da8cb8-cfdb-11e7-8ffd-d800555694fd.png)

**Comment by aloneguid** [comment]
I can confirm that no values in the repeatable field causes problems though ;)

**Comment by aloneguid** [comment]
found the problem now, the fix should come next week )

**Comment by aloneguid** [comment]
Fixed in 2.0 preview ;) You can check if it work for you by referencing a cusom nuget feed for dev releases: https://ci.appveyor.com/nuget/parquet-dotnet-nvq9lkymhrek
